File: frontend/src/api/scheduleScoreReport.ts
import request from '../utils/request'
import type { ApiResponse, PageResult } from './types'

export interface ScheduleScoreReport {
  id: number
  semesterId: number
  score: number
  grade: string
  gradeName: string
  conflictCount: number
  unfinishedTaskCount: number
  teacherOverloadCount: number
  classOverloadCount: number
  fridayAfternoonCount: number
  deductionDetail: string
  suggestion: string
  createTime: string
}

export interface ScheduleScoreResult {
  score: number
  grade: string
  gradeName: string
  conflictCount: number
  unfinishedTaskCount: number
  teacherOverloadCount: number
  classOverloadCount: number
  fridayAfternoonCount: number
  deductionDetail: string[]
  suggestion: string[]
}

export function generateScheduleScore(semesterId?: number) {
  return request.post<ApiResponse<ScheduleScoreResult>>('/schedule-score/generate', {}, { params: { semesterId } })
    .then((r) => {
      if (!r.data) {
        throw new Error('响应数据为空')
      }
      return r.data.data
    })
}

export function getLatestScheduleScore(semesterId?: number) {
  return request.get<ApiResponse<ScheduleScoreReport>>('/schedule-score/latest', { params: { semesterId } })
    .then((r) => {
      if (!r.data) {
        throw new Error('响应数据为空')
      }
      return r.data.data
    })
}

export function getScheduleScoreHistory(params: {
  semesterId?: number
  grade?: string
  startTime?: string
  endTime?: string
  page?: number
  size?: number
}) {
  return request.get<ApiResponse<PageResult<ScheduleScoreReport>>>('/schedule-score/reports', { params })
    .then((r) => {
      if (!r.data) {
        throw new Error('响应数据为空')
      }
      return r.data.data
    })
}

File: backend/src/main/java/com/paike/scheduler/entity/ScheduleScoreReport.java
package com.paike.scheduler.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@TableName("schedule_score_report")
public class ScheduleScoreReport {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long semesterId;

    private Integer score;

    private String grade;

    private Integer conflictCount;

    private Integer unfinishedTaskCount;

    private Integer teacherOverloadCount;

    private Integer classOverloadCount;

    private Integer fridayAfternoonCount;

    @TableField("deduction_detail")
    private String deductionDetail;

    private String suggestion;

    private LocalDateTime createTime;

    private String gradeName;
}
